includes/class-sensei-frontend.php

```diff
@@ -1591,8 +1591,16 @@ function sensei_handle_login_request( ) {
 
 		    	// check if the requests login is an email address
 		    	if( is_email(  trim( $_REQUEST['log'] ) )  ){
+		    		$login = sanitize_email( $_REQUEST['log'] );
+
 		    		// query wordpress for the users details
-		    		$user =	get_user_by( 'email', sanitize_email( $_REQUEST['log'] )  );
+		    		$user = get_user_by( 'email', $login );
+
+		    		if( ! $user ) {
+		    			// Occasionally a user's username IS an email,
+		    			// but they have changed their actual email, so check for this case.
+		    			$user = get_user_by( 'login', $login );
```

Comment:
@jtsternberg 

If we wanted to give priority of login over email, as you state in your comment, this should be checked first